### ISSUE TITLE
dulwich.porcelain.add() corrupts branch

### ISSUE THREAD

**Issue by tillias** [jelmer/dulwich#1178]
Hello,

I have a repo inside c:/tmp/release-package-v0.0.1

When I run 

```
local_path = 'c:/tmp/release-package-v0.0.1'
porcelain.add(local_path, local_path)
rst = status(local_path)
print(rst)
```

It shows some strange status:

`GitStatus(staged={'add': [b'./'], 'delete': [], 'modify': []}, unstaged=[], untracked=['LICENSE', 'README.md', 'synchronizer.py'])`

So my untracked files are not added to repo but some b'./' instead.

If I commit those changes, then git repository becomes corrupted and I can't run 'status' or standalone git commands inside it anymore.

It says 'Fatal: empty filename in tree'

Please help me how to add new files into repository, any working workaround will be much appreciated! It seems problem is with **relpath** in porcelain.py but I'm not sure how to fix it :(

```
        for p in paths:
            path = Path(p)
            relpath = str(path.resolve().relative_to(repo_path))
            # FIXME: Support patterns
```

Thank you!

**Comment by tillias** [comment]
Or here is source code: https://github.com/tillias/aws-code-commit-sync/blob/38366c18c45fdaeec898032a6eced7edf5e4f6e7/src/git_helper.py#L32

Please help, I'm completely stuck with it.

**Comment by jelmer** [comment]
> local_path = 'c:/tmp/release-package-v0.0.1'

This is an odd local path - it's not a valid windows path IIUC, which should be using backslashes.

Any reason why you're passing in local_path twice, rather than just once? Does it work if you just specify it once, and with backslahses?

Dulwich should normally just scan directories recursively but not explicitly add them. This is exactly what Repo.stage() does.

**Comment by tillias** [comment]
No, it doesn't work if I pass it only once or specify another valid folder. That is exactly what I'm trying to solve, it looks like current version of porcelain uses current working path (e.g. where caller script is located) instead of parameter I specify in porcelain.add()

I want to start script in one folder and pass another which I want to add into git repository. 

E.g. script is located in /c/scipt_folder/script_using_porcelain_add.py and my repo is in /c/some_git_repo_i_want_to_add

If I call porcelain.add() with any combination of parameters (I really tried hard with all possible combinations before opening the ticket) it always adds /c/scipt_folder/

Please note that  /c/scipt_folder/ is also versioned in git
